spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/SparkTestBase.java

```diff
@@ -118,6 +119,8 @@ protected long waitUntilAfter(long timestampMillis) {
             return row.getList(pos);
           } else if (value instanceof scala.collection.Map) {
             return row.getJavaMap(pos);
+          } else if (value.getClass().isArray() && value.getClass().getComponentType().isPrimitive()) {
+            return IntStream.range(0, Array.getLength(value)).mapToObj(i -> Array.get(value, i)).toArray();
```

Comment:
Question: This change to the test class seems somewhat complicated relative to the other cases. Is there any way to check for `value instanceof scala.collection.Array` (assuming that Scala types are returned) like in the map case and in the basic list case?

I guess my question is, why do we need this block when we have the check for `scala.collection.Seq` already. Can you help me understand?

And possibly, since it is a bit more complicated, a comment addressing what case this is expected to handle in the code might be called for.

Reply:
yeap, i’ll simplify the code and add instructions

Reply:
What does this convert the byte array to? Why does it need to change? I thought `byte[]` was fine to return.

Reply:
The `assertEquals` method cannot verify whether binary arrays are equal.  If we do not convert binary arrays to Object[], we may need to create a separate method to compare binary arrays

Reply:
This representation should not change for tests. I think that the original type is correct. If you need to fix tests that are failing because of array comparison, then you should update the assertions instead of runtime code.

Reply:
okey